src/html.cr

```diff
@@ -1,10 +1,24 @@
 module HTML
   SUBSTITUTIONS = {
-    '&'  => "&amp;",
-    '\"' => "&quot;",
-    '\'' => "&apos;",
-    '<'  => "&lt;",
-    '>'  => "&gt;",
+    '!'      => "&#33;",
+    '"'      => "&quot;",
+    '$'      => "&#36;",
+    '%'      => "&#37;",
+    '&'      => "&amp;",
+    '\''     => "&#39;",
+    '('      => "&#40;",
+    ')'      => "&#41;",
+    '='      => "&#61;",
+    '>'      => "&gt;",
+    '<'      => "&lt;",
+    '+'      => "&#43;",
+    '@'      => "&#64;",
+    '['      => "&#91;",
+    ']'      => "&#93;",
+    '`'      => "&#96;",
+    '{'      => "&#123;",
+    '}'      => "&#125;",
+    '\u{a0}' => "&nbsp",
```

Comment:
The `;` is not needed for it to work. Do you want me to add it, for consistency purpose ?